array.c

```diff
@@ -7843,13 +7843,20 @@ rb_ary_deconstruct(VALUE ary)
  *
  *  == Creating Arrays
  *
- *  A new array can be created by using the literal constructor
- *  <code>[]</code>.  Arrays can contain different types of objects.  For
+ *  You can create an \Array object explicitly with:
+ *
+ *  - An {array literal}[doc/syntax/literals_rdoc.html#label-Array+Literals].
```

Comment:
RDoc is not only for HTML format, should be rdoc-links.

Reply:
This style is used in a lot of recent documentation, because people aren't sure how to use an rdoc-link to hyperlink to a specific heading in the generated documentation.  Can you show an example of using an rdoc-link to generate the same link? Maybe it is possible with `rdoc-label`?

Reply:
Basically, this comes down to these two usages:

```
 *  You can create an \Array object explicitly with:
 *
 *  - An {array literal}[doc/syntax/literals_rdoc.html#label-Array+Literals].
 *
 *  You can convert certain objects to Arrays with:
 *
 *  - \Method {Array}[Kernel.html#method-i-Array].
 *
```

I have not found any form for these links that works, other than the above.  If someone knows how to write these links without `.html`,  please provide examples.

Specifically, what's needed is:
- Link to an instance method in `Kernel`.
- Link to a section in `doc/syntax/literals.rdoc`.

Reply:
A link to an instance method Kernel would be `Kernel#method_name`.  So those should be changed.  In the example above, switch to:

```
* - \Method Kernel#Array
```

Similar changes for other method links.

Linking to section headings can be done with `@`, according to the rdoc documentation:

```
A heading inside the documentation can be linked by following the class or method by an @ then the heading name. RDoc::Markup@Links will link to this section like this: Links at RDoc::Markup. Spaces in headings with multiple words must be escaped with + like RDoc::Markup@Escaping+Text+Markup. Punctuation and other special characters must be escaped like CGI.escape.
```

That shows how to link to the section heading of a class. However, I'm not sure how to generate a link to a section heading of a file.  Maybe `{array literal}[doc/syntax/literals.rdoc@Array+Literals]` works?

Reply:
I've tried many things, including these:

- `{Array}[Kernel#Array]` did not work; generated `<a href="Kernel#Array">Array</a>`. which did not work.
- `{array literal}[doc/syntax/literals.rdoc@Array+Literals]`; generated `<a href="doc/syntax/literals.rdoc@Array+Literals">array literal</a>`, which also did not work.

Also, various forms with `#method-i-Array` did not work.

Could it be that things are different for a module such as `Kernel`?  And for a file such as `literals.rdoc`?

Reply:
Does `Kernel#Array` by itself work?  If so, we should switch all instance method references to that style.  It's certainly better than the current `{Array}[Kernel.html#method-i-Array]`.

It's possible rdoc does not currently support linking to headings in files, or linking to headings inside `[]`.  So maybe nothing can currently be done regarding links such as `{array literal}[doc/syntax/literals_rdoc.html#label-Array+Literals]` until rdoc adds support for it.

Reply:
No, `Kernel#Array` does not work.  Maybe we need to hear from @nobu on this.

Reply:
I'm asking @aycabta to fix RDoc.
The links are OK for now.

Reply:
Thanks, @nobu.  When RDoc is fixed, I will sweep the major classes free of these.

Reply:
> Link to a section in `doc/syntax/literals.rdoc`.

I think it would be better to use `foo.rdoc#label-bar` for labeled links to `foo.rdoc` instead of `foo_rdoc.html#label-bar`.

The `label-` part is a result of RDoc's internal processing, so it's probably not a good idea either, but if RDoc changes `foo.rdoc#bar` to `foo_rdoc.html#label-bar`, the part originally written as `abc.rdoc#label-def` will now be  `abc_rdoc.html#label-label-def`. There are many places like this in the existing documentation. Even if other new notations are introduced, careful consideration is needed to ensure appropriate specifications.

> Link to an instance method in `Kernel`.

I fixed it by https://github.com/ruby/rdoc/pull/853. You can use `Kernel#String`.

Reply:
Thanks, @aycabta.  If I change to Kernel#String, I won't be able to verify in Ruby 3.0.2.  If I put the 8 changes in a PR, can you verify?